tracing: remove DeprecatedStats field from RecordedSpan

yuzefovich opened the pull request:
In 21.1 cycle we changed the tracing infrastructure as well as the way
we propagate execution statistics. With the latter change we bumped the
DistSQL version in order to prohibit the mixed-version clusters
scenario; however, with the former we kept the old stats field in
`RecordedSpan` and renamed it to `DeprecatedStats` to be removed in 21.2
cycle. I believe that we don't need that deprecated field at all given
the DistSQL version bump, so this commit removes the deprecated stats
field entirely.

Release justification: low-risk update to existing functionality.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61352)
<!-- Reviewable:end -->

yuzefovich committed:
tracing: remove DeprecatedStats field from RecordedSpan

In 21.1 cycle we changed the tracing infrastructure as well as the way
we propagate execution statistics. With the latter change we bumped the
DistSQL version in order to prohibit the mixed-version clusters
scenario; however, with the former we kept the old stats field in
`RecordedSpan` and renamed it to `DeprecatedStats` to be removed in 21.2
cycle. I believe that we don't need that deprecated field at all given
the DistSQL version bump, so this commit removes the deprecated stats
field entirely.

Release justification: low-risk update to existing functionality.

Release note: None

yuzefovich commented:
TFTR!

bors r+
